notebooks/E2Ev2.ipynb

```diff
@@ -0,0 +1,350 @@
+    "    # Rückgabe des ersten Wertes in dem bestimmten Datenset\n",
+    "    all_values_426.append(qa.get_value1_by_year(company_id=i, year=year))\n",
+    "    print(\"Value 1 = \" + qa.get_value1_by_year(company_id=i, year=year))\n",
+    "\n",
+    "    # Erhalten der Datenreferenz in Bytes, wenn eine Datenquelle hinterlegt ist\n",
+    "    all_datasource_reference_bytes.append(qa.get_datasource_reference_bytes(company_id=i, year=year))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for o in all_datasets:\n",
+    "    print(o)\n",
+    "    print(\"\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2. Laden der Datenquelle von Dataland und Überführen in Textform"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finden des relevanten Bereichs im PDF-Dokument"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import io\n",
+    "\n",
+    "import pypdf  # type: ignore\n",
+    "\n",
+    "all_page_number = []\n",
+    "all_file_id = []\n",
+    "all_file_name = []\n",
+    "all_tag_name = []\n",
+    "all_page = []\n",
+    "\n",
+    "for t in all_datasets:\n",
+    "    dataset_section426 = t.data.general.general.nuclear_energy_related_activities_section426\n",
+    "    all_page_number.append(dataset_section426.data_source.page)\n",
+    "\n",
+    "    all_file_id.append(dataset_section426.data_source.file_reference)\n",
+    "    all_file_name.append(dataset_section426.data_source.file_name)\n",
+    "    all_tag_name.append(dataset_section426.data_source.tag_name)\n",
+    "    all_page.append(dataset_section426.data_source.page)\n",
+    "\n",
+    "    pdf = dataland_client.documents_api.get_document(dataset_section426.data_source.file_reference)\n",
+    "    pdf_stream = io.BytesIO(pdf)\n",
+    "    pdf_reader = pypdf.PdfReader(pdf_stream)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for u in all_page_number:\n",
+    "    print(u)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dataland_qa_lab.dataland.data_extraction as qa_lab\n",
+    "\n",
+    "all_values_426_pdf = []\n",
+    "\n",
+    "i = 0\n",
+    "for z in all_datasets:  # noqa: B007\n",
+    "    relevant_page_number = int(all_page_number[i])\n",
+    "\n",
+    "    relevant_page = qa_lab.get_relevant_page_of_pdf(relevant_page_number, pdf_reader)\n",
+    "\n",
+    "    text_of_page = qa_lab.extract_text_of_pdf(relevant_page)\n",
+    "\n",
+    "    section_426 = qa_lab.extract_section_426(text_of_page)\n",
+    "    all_values_426_pdf.append(section_426)\n",
+    "\n",
+    "    print(section_426)\n",
+    "\n",
+    "    i += 1"
```

Comment:
Tip: If you need the index when using a python for loop, consider using `enumerate`:
```python
for i, dataset in enumerate(all_datasets):
```